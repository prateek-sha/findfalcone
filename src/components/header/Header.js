import React, { useContext } from "react";
import { Grid, makeStyles, Typography } from "@material-ui/core";
import { useHistory, Link } from "react-router-dom";
import { globalContext } from "../../context/ContextProvider";
import Fade from "react-reveal/Fade";

const useStytles = makeStyles({
  root: {
    color: "white",
  },
  option: {
    padding: "2em",
    float: "left",
  },
  heading: {
    alignItems: "center",
    textAlign: "center",
  },
  link: {
    color: "white",
    textDecoration: "none",
    cursor: "pointer",
  },
  margin: {
    marginLeft: ".5em",
  },
});

export const Header = () => {
  const { Reset } = useContext(globalContext);
  const classes = useStytles();
  const history = useHistory();

  const handleReset = () => {
    Reset();
    history.push("/");
  };
  return (
    <Grid className={classes.root} container direction="column">
      <Fade right>
        <Grid
          className={classes.option}
          container
          direction="row"
          justify="flex-end"
        >
          <Grid className={classes.link} item onClick={handleReset}>
            Reset
          </Grid>
          <Grid className={classes.margin} item>
            |
          </Grid>
          <Grid className={classes.margin} item>
            <a className={classes.link} href="https://www.geektrust.in/">
              Geek Trust Home
            </a>
          </Grid>
        </Grid>
      </Fade>
      <Grid className={classes.heading} item>
        <Fade left>
          <Typography variant="h2">Finding Falcone!</Typography>
        </Fade>
      </Grid>
    </Grid>
  );
};
