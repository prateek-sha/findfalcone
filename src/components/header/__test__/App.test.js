import React from "react";
import { Header } from "../Header";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";

it("matches snapshot ", () => {
  const tree = renderer.create(<Header></Header>).toJSON();
  expect(tree).toMatchSnapshot();
});
