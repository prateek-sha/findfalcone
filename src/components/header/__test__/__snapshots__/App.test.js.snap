// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot  1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-direction-xs-column"
>
  <div
    className="react-reveal"
    style={
      Object {
        "opacity": undefined,
      }
    }
  >
    <div
      className="MuiGrid-root makeStyles-option-2 MuiGrid-container MuiGrid-justify-xs-flex-end"
    >
      <div
        className="MuiGrid-root makeStyles-link-4 MuiGrid-item"
        onClick={[Function]}
      >
        Reset
      </div>
      <div
        className="MuiGrid-root makeStyles-margin-5 MuiGrid-item"
      >
        |
      </div>
      <div
        className="MuiGrid-root makeStyles-margin-5 MuiGrid-item"
      >
        <a
          className="makeStyles-link-4"
          href="https://www.geektrust.in/"
        >
          Geek Trust Home
        </a>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root makeStyles-heading-3 MuiGrid-item"
  >
    <div
      className="react-reveal"
      style={
        Object {
          "opacity": undefined,
        }
      }
    >
      <h2
        className="MuiTypography-root MuiTypography-h2"
      >
        Finding Falcone!
      </h2>
    </div>
  </div>
</div>
`;
